### ISSUE TITLE
bug(linter) incorrect fixer for `no-useless-escape`

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#5227]
```ts
const regex = /(https?:\/\/github\.com\/(([^\s]+)\/([^\s]+))\/([^\s]+\/)?(issues|pull)\/([0-9]+))|(([^\s]+)\/([^\s]+))?#([1-9][0-9]*)($|[\s\:\;\-\(\=])/;
```

is being fixed to 
```ts
const regex = /(https?:\/\/github\.com\/(([^\s]+)\/([^\s]+))\/([^\s]+\/)?(issues|pull)\/([0-9]+))|(([^\s]+)\/([^\s]+))?#([1-9][0-9]*)($|[\s:;-(=])/;
```

which is an invalid regex
```

  × Invalid regular expression: Character class range out of order
   ╭─[./src/vs/workbench/api/test/browser/extHostDocumentData.test.ts:2:142]
 1 │ const regex = /(https?:\/\/github\.com\/(([^\s]+)\/([^\s]+))\/([^\s]+\/)?(issues|pull)\/([0-9]+))|(([^\s]+)\/([^\s]+))?#([1-9][0-9]*)($|[\s:;-(=])/;
   ·                                                                                                                                              ──
   ╰────

Finished in 3ms on 1 file with 93 rules using 12 threads.
Found 0 warnings and 1 error.
```

**Comment by camc314** [comment]
looks like eslint doesn't report an error for this case (escacpe on the `\-`), but we do.

gonna leave the fixer alone and fix this false positive

oxlint >> https://oxc-project.github.io/oxc/playground/?code=3YCAAICggICAgICAgICxG0qZRraXTnShZDZHIj18ztvE%2FrAEWU6BK3tu8l%2BruCZLm6bBN39t5QCA

eslint >> https://eslint.org/play/#eyJ0ZXh0IjoiY29uc3QgcmVnZXgwID0gL1tcXHNcXDpcXDtcXC1cXChcXD1dLzsiLCJvcHRpb25zIjp7InJ1bGVzIjp7Im5vLXVzZWxlc3MtZXNjYXBlIjpbImVycm9yIl19LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==


<img width="1914" alt="Screenshot 2024-08-26 at 15 43 11" src="https://github.com/user-attachments/assets/3d1227a9-1fc6-43f9-9fc0-4978b0b5de3c">


minimal test case:
```
/[\s\-(]/;
```

**Comment by Boshen** [comment]
> I think it's going to be better if we wait till the linter has access to the pared regex ast and do it that way, else it's just going to be duplicating some of the regex parsing code

We do now 😁

**Comment by camchenry** [comment]
This should be fixed now with https://github.com/oxc-project/oxc/pull/5974 merged and I added a regression test for it here too: https://github.com/oxc-project/oxc/pull/5975
